oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java

```diff
@@ -635,6 +643,101 @@ public void filterWhenRequestAttributesNotSetAndHooksInitHooksResetThenDefaultsN
 		assertThat(getBody(request)).isEmpty();
 	}
 
+	// gh-7228
+	@Test
+	public void publisherWhenHooksInitAndSubscribedOutsideOfWebContextThenShouldNotThrowNpe() throws Exception {
+		Mono<Context> publisher = Mono.subscriberContext();
+		this.function.afterPropertiesSet();			// Hooks.onLastOperator() initialized
+
+		publisher.block();
+	}
+
+	// gh-7228
+	@Test
+	public void publisherWhenHooksInitAndSubscribedOutsideOfWebContextAndOutsideOfSecurityContextAndWithParentContextThenShouldNotReplaceParentContextAndShouldNotAddRequestContext() throws Exception {
```

Comment:
I'm not sure about this test and the next one. It's quite low level and not directly testing the API of `ServletOAuth2AuthorizedClientExchangeFilterFunction`. Most tests are directly testing `filter()` and  `defaultRequest()`, whereas these 2 tests are directly using the Reactor API's.

The preference would be to change these tests so they are aligned with the existing `filter()` tests.

Reply:
The main issue was that current instrumentation affects all application   
So this tests check if it is possible to performe operations on Publishers without web/sucurity context with initialized reactor instrumentation (which is ServletOAuth2AuthorizedClientExchangeFilterFunction functionality)

Reply:
Both of these tests do not directly test the API of `ServletOAuth2AuthorizedClientExchangeFilterFunction` but instead are testing internal implementation details. In order to simplify the tests, please make `RequestContextSubscriber` package-private along with the constructor. This will allow you to test the constructed state of `RequestContextSubscriber` using `currentContext()`. Please remove these 2 tests after you have added the new ones. Thanks.

Reply:
as this PR fixes scenario without web/security context (see https://github.com/spring-projects/spring-security/pull/7235#discussion_r316789550 ) and in this scenario `ExchangeFilterFunction` API of `ServletOAuth2AuthorizedClientExchangeFilterFunction` is not used  
but `ServletOAuth2AuthorizedClientExchangeFilterFunction#afterPropertiesSet` is used and it is part of API it is makes sense to test these scenarios without usage of `ExchangeFilterFunction` part of API.  

Other scenarios with direct usage of `ExchangeFilterFunction` API are covered by existing tests. 
I will make `RequestContextSubscriber` package-private but test will not change very much as they could not use `ExchangeFilterFunction` API by scenario.   

These test could be moved to a new test class but I do not see point in this as they still test `ServletOAuth2AuthorizedClientExchangeFilterFunction#afterPropertiesSet` API.